Scripts/GeneralTools/veleroBackupASEA/dr.sh

```diff
@@ -0,0 +1,724 @@
+#!/bin/bash
+
+# Reconfigure DR for AS in AKS/EKS
+# Applicable ASEA version: 2023.10.0-2023.10.7, 2024.10.0-2024.10.1
+#
+# For above mentioned version, uipathctl snapshot command doesn't create backup of volume data. This script mitigate the backup issue by reinstalling the velero and argocd configuration.
+#
+# Script performs below operations:
+# - Re-install velero with node-agent enabled. Node-agent is required to create snapshot of file based PV
+# - Updates argocd to ignore labels and annotations of deployment, statefulset and PVC resources.
+# - Updates insight PVC labels to enable backup of insight data [if insight is installed]
+# - Updates insight statefulset resource to enable file-based snapshot of insight data [if insight is installed]
+#
+# Note:
+# - Before executing the script, ensure backup store is configured
+# - Re-installing argocd, insight, velero and studioweb components will override the modified configuration. Please re-run this script post re-installation of these components.
+
+set -ueo pipefail
+
+NS_ARGOCD="${NS_ARGOCD:=argocd}"
+NS_UIPATH="${NS_UIPATH:=uipath}"
+NS_ISTIO="${NS_ISTIO:=istio-system}"
+NS_AIRFLOW="${NS_AIRFLOW:=airflow}"
+NS_MONITORING="${NS_MONITORING:=monitoring}"
+
+REGISTRY=
+REPOSITORY=
+PLUGIN_NAME=
+PLUGIN_IMAGE=
+KUBECTL_IMAGE_TAG=
+KUBECTL_IMAGE_REPO=
+PROVIDER=
+BUCKET=
+AZURE_RG=
+AZURE_ACCOUNT=
+AZURE_SUBSCRIPTION_ID=
+AWS_REGION=
+AWS_ARN=
+CREDENTIAL_SECRET_SET="false"
+CREDENTIAL_SECRET_NAME="velero-azure"
+VELERO_VERSION=
+VELERO_CHART_VERSION=
+VELERO_NODE_TOLERATIONS=
+
+# override image if required
+HELM_INSTALLER_IMAGE_TAG=
```

Comment:
shouldn't all these variable allow override ? via `HELM_INSTALLER_IMAGE_TAG=${HELM_INSTALLER_IMAGE_TAG:=""}`

Reply:
not required since we are fetching it from installed release. This script is intended for reinstallation of velero. It doesn't support installing velero fresh.